src/librustdoc/lib.rs

```diff
@@ -651,6 +651,22 @@ fn main_options(options: config::Options) -> MainResult {
     let config = core::create_config(options);
 
     interface::create_compiler_and_run(config, |compiler| {
+        let sopts = &compiler.session().opts;
+        if sopts.describe_lints {
+            let mut lint_store = rustc_lint::new_lint_store(
+                sopts.debugging_opts.no_interleave_lints,
+                compiler.session().unstable_options(),
+            );
+            let registered_lints =
+                if let Some(register_lints) = compiler.register_lints() {
+                    register_lints(compiler.session(), &mut lint_store);
+                    true
+                } else {
+                    false
+                };
+            describe_lints(compiler.session(), &lint_store, registered_lints);
+            return Ok(());
+        }
         compiler.enter(|queries| {
             let sess = compiler.session();
```

Comment:
This is more complicated than it needs to be (it looks like you copied it from `rustc_driver`, which runs this before it registers lints).
```suggestion
        let sess = compiler.session();
        if sess.opts.describe_lints {
            let (_, lint_store) = &*compiler.register_plugins()?.peek();
            describe_lints(sess, &lint_store, true);
            return Ok(());
        }
        compiler.enter(|queries| {
```

Reply:
Yes, you're right—I did just pillage from `rustc_driver`.  This is much better!  Thanks